src/openms/source/QC/Suitability.cpp

```diff
@@ -0,0 +1,316 @@
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Tom Waschischeck $
+// $Authors: Tom Waschischeck $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/ID/FalseDiscoveryRate.h>
+#include <OpenMS/CONCEPT/Constants.h>
+#include <OpenMS/CONCEPT/LogStream.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <OpenMS/METADATA/PeptideIdentification.h>
+#include <OpenMS/METADATA/PeptideHit.h>
+#include <OpenMS/QC/Suitability.h>
+
+using namespace std;
+
+namespace OpenMS
+{
+  Suitability::Suitability()
+    : DefaultParamHandler("Suitability"), results_{}
+  {
+    defaults_.setValue("no_re_rank", "false", "Enable/Disable re-ranking");
+    defaults_.setValidStrings("no_re_rank", ListUtils::create<String>("true,false"));
+    defaults_.setValue("novo_fract", 1., "Fraction of how many cases, where a de novo peptide scores just higher than the database peptide, will be re-rank");
+    defaults_.setMinFloat("novo_fract", 0.);
+    defaults_.setMaxFloat("novo_fract", 1.);
+    defaults_.setValue("FDR", 0.01, "Filtering peptide hits based on this q-value");
+    defaults_.setMinFloat("FDR", 0.);
+    defaults_.setMaxFloat("FDR", 1.);
+    defaultsToParam_();
+  }
+  
+  void Suitability::computeSuitability(vector<PeptideIdentification>& pep_ids)
+  {
+    bool no_re_rank = param_.getValue("no_re_rank").toBool();
+    double novo_fract = param_.getValue("novo_fract");
+    double FDR = param_.getValue("FDR");
+
+    SuitabilityData d;
+
+    bool fdr_done = false;
+    bool q_value_score = false;
+    if (pep_ids[0].getScoreType() == "q-value")
```

Comment:
.. because then you can simply leave the function here 
`if (pep_ids.empty) return;` to avoid a segfault when checking the first element